Removed UMAGetSystemVersion to fix runtime warning on OS X >= 10.10

TcT2k opened the pull request:
On OS X 10.10 a call to UMAGetSystemVersion produces a warning on the console at runtime:

WARNING: The Gestalt selector gestaltSystemVersion is returning 10.9.4 instead of 10.10.4. Use NSProcessInfo's operatingSystemVersion property to get correct system version number.*

Many calls guarded by checks with UMAGetSystemVersion where guarding < 10.7.
As the minimum version for 3.1.x is 10.7 I've simply removed that occurances.

Other occurances have been replaced by wxPlatfromInfo::CheckOSVersion which internally uses wxGetOSVersion, which uses Gestalt calls not producing the runtime warning.

vadz commented:
Thanks, this looks good but ideal would be to split this in 2 commits: first one removing support for OS X < 10.7 and the second one replacing the remaining occurrences of `UMAGetSystemVersion()`. I'll try to do it myself later if you can't but I'd prefer to do this before committing.

TcT2k commented:
I can split that in two commits.

TcT2k committed:
Removed OS X code targeted at versions prior to OS X 10.7

TcT2k commented:
Changes are now split in 2 commits like suggested.

TcT2k committed:
Removed UMAGetSystemVersion to fix runtime warning on OS X 10.10 and newer

On OS X 10.10 a call to UMAGetSystemVersion produces a warning on the console at runtime:
WARNING: The Gestalt selector gestaltSystemVersion is returning 10.9.4 instead of 10.10.4. Use NSProcessInfo's operatingSystemVersion property to get correct system version number.*

Calls to UMAGetSystemVersion have been replaced by wxPlatformInfo::CheckOSVersion which internally uses wxGetOSVersion, which uses Gestalt calls not producing the runtime warning.
